### PR TITLE
fix(@schematics/angular): add "type" option in enum schematic

### PR DESCRIPTION
**Pull request by mephistorine**
closes: #20667

### CODE REVIEW

**Review by alan-agius4 — CHANGES_REQUESTED** [review]
Thanks for this.

Couple of things

- Can you please drop the first commit?
- Kindly use `@schematics/angular` as commit message scope.
- Add `closes: #20667` as commit message footer.

**Comment by mephistorine** [comment]
@alan-agius4 thanks, ok.

**Review by alan-agius4 — APPROVED** [review]

File: packages/schematics/angular/enum/index.ts [hunk 1]
```diff
@@ -33,6 +33,8 @@ export default function (options: EnumOptions): Rule {
     options.name = parsedPath.name;
     options.path = parsedPath.path;
 
+    options.type = !!options.type ? `.${options.type}` : '';
```
Comment (alan-agius4) [inline packages/schematics/angular/enum/index.ts:36]: nit
```suggestion
    options.type = options.type ? `.${options.type}` : '';
```

File: packages/schematics/angular/enum/schema.json [hunk 2]
```diff
@@ -28,6 +28,11 @@
         "$source": "projectName"
       }
     },
+    "type": {
+      "type": "string",
+      "description": "Adds a developer-defined type to the filename, in the format \"name.type.ts\".",
+      "default": ""
```
Comment (alan-agius4) [inline packages/schematics/angular/enum/schema.json:34]: nit 
```suggestion
```

File: packages/schematics/angular/enum/index.ts (hunk 1 above)
Comment (mephistorine) [reply packages/schematics/angular/enum/index.ts]: done 👍🏻

File: packages/schematics/angular/enum/schema.json (hunk 2 above)
Comment (mephistorine) [reply packages/schematics/angular/enum/schema.json]: done 👍🏻

**Review by alan-agius4 — APPROVED** [review]
LGTM, thanks for your contribution.

Can you kindly squash the commits? Thanks.

**Comment by mephistorine** [comment]
@alan-agius4 yes, give a second. :)

**Review by alan-agius4 — APPROVED** [review]
LGTM, thanks

**Comment by mephistorine** [comment]
@alan-agius4 Thanks a lot 😃

**Pull request merged**